site/v5/testing/testing.md

```diff
@@ -0,0 +1,96 @@
+- **BeforeEach method(automatically called):** Resets mongock to allow re-utilization(not recommended in production) and build the runner
+- **AfterEach method(automatically called):** Cleans both Mongock repositories(lock and migration) 
+- **Dependency injections:** It ensures the required dependencies(Mongock builder, connectionDriver, etc.) are injected 
+- **executeMongock() method:** To perform the Mongock migration
+- **@TestPropertySource(properties = {"mongock.runner-type=NONE"}):** To prevent Mongock from injecting(and automatically executing) the Mongock runner bean. This is important to allow multiple Mongock runner's executions.
+
+Please follow these steps...
+### 1. Import the `mongock-springboot-junit5` dependency to your project
+Assuming you have imported already `mongock-springboot` to you project, you only need to add
+```xml
+        <dependency>
+            <groupId>io.mongock</groupId>
+            <artifactId>mongock-springboot-junit5</artifactId>
+            <scope>test</scope>
+        </dependency>
+```
+
+### 2. Add the additional dependencies to your project
+You probably need `Springboot starter test`, `JUnit5`, `Testcontainers`...
+
+### 3. Database initialization. 
+
+Although there are multiple ways of doing this, we present what we think provides a good balance easy-flexible
+<p class="text-center">
+    <img src="/images/integration-test-springboot-junit5-db-initialization.png" alt="ChangeUnit unit test">
+</p>
+
+### 4. Create the test class extending the `MongockSpringbootJUnit5IntegrationTestBase`
+
+This class, in addition to extending `MongockSpringbootJUnit5IntegrationTestBase`, it should also bring the database initialization and the application environment.
+<br /><br />
+This is an example.
+<p class="text-center">
+    <img src="/images/integration-test-springboot-junit5-test-class.png" alt="Integration test with JUnit5">
+</p>
+
+
+
+## Integration test with Springboot Runner WITHOUT Junit5
+
+In this case Mongock provides pretty much the same than in the JUnit5 case, with the exception of the `before` and `after` methods, what forces you to make these calls explicitly.
+
+Based on the previous scenario, the relevant modifications are
+1. Import the dependency `mongock-springboot-test` instead `mongock-springboot-junit5`
+1. Extend from `MongockSpringbootIntegrationTestBase` instead `MongockSpringbootJUnit5IntegrationTestBase`
+2. Explicitly call the methods `super.mongockBeforeEach()` and `super.mongockAfterEach()`
+
+The test class should look like this
+<p class="text-center">
+    <img src="/images/integration-test-springboot-test-class.png" alt="Integration test without JUnit5">
+</p>
+
+
+## Integration test with Standalone Runner
+In this case it requires much less help to develop integration tests, as the Standalone runner provides more control over the process.
```

Comment:
Remove entire paragraph.. I don't think it adds a lot of value - or just leave "the Standalone runner provides more control over the process, simplifying the testing complexity"

Reply:
What I am trying to say here is that, while for springboot you NEED some utility classes and "help", with the standalone you don't because you have more control over it. 

I reformulate to:

```
The standalone runner provides more control over the process, allowing you to implement integration 
tests without the need of additional support.

You need to take into account the following

- Mongock Runner cannot be execute multiple times, so you need to build a new runner instance 
and execute it in every test execution.

- ConnectionDriver cannot be reused, meaning you need to create a new ConnectionDriver instance 
in every test execution and provide it to the Mongock builder

- If you are sharing the same database for multiple tests, make sure you clean the database, in case
 you want to start fresh each test case.

```